misc/toml-parser/src/main/java/io/ballerina/toml/semantic/ast/TomlTransformer.java

```diff
@@ -321,8 +331,7 @@ private TomlKeyNode getTomlKeyNode(SeparatedNodeList<ValueNode> identifierList)
             TomlBasicValueNode transformedNode = (TomlBasicValueNode) node.apply(this);
             nodeList.add(new TomlKeyEntryNode(transformedNode));
         }
-
-        return new TomlKeyNode(nodeList);
+        return new TomlKeyNode(nodeList, getTomlNodeListLocation(new ArrayList<>(nodeList)));
```

Comment:
Shall we avoid copying the list to a new list?
```suggestion
        return new TomlKeyNode(nodeList, getTomlNodeListLocation(nodeList));
```